File: apollo/src/main/java/io/muun/apollo/domain/model/SignupDraft.java
package io.muun.apollo.domain.model;


import io.muun.apollo.data.serialization.SerializationUtils;
import io.muun.apollo.external.Globals;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;

import javax.annotation.Nullable;
import javax.validation.constraints.NotNull;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
public class SignupDraft {

    public static SignupDraft deserialize(String json) {
        return SerializationUtils.deserializeJson(SignupDraft.class, json);
    }

    @NotNull
    public Integer versionCode = Globals.INSTANCE.getVersionCode();

    @Nullable
    public Boolean isExistingUser;

    @NotNull
    public SignupStep step = SignupStep.START;

    @NotNull
    public Boolean canUseRecoveryCode;

    @Nullable
    @JsonIgnore
    public transient String password;

    @Nullable
    public String email;

    public String serialize() {
        return SerializationUtils.serializeJson(SignupDraft.class, this);
    }
}

File: apollo/src/main/java/io/muun/apollo/domain/model/SignupDraft.kt
package io.muun.apollo.domain.model

import io.muun.apollo.data.external.Globals

/**
 * We need empty constructors to work with jackson de/serialization.
 * On the JVM, if the generated class needs to have a parameterless constructor, default values
 * for all properties have to be specified. See:
 * https://kotlinlang.org/docs/reference/classes.html#constructors
 */
class SignupDraft(
    var versionCode: Int = Globals.INSTANCE.versionCode,
    var isExistingUser: Boolean = false,
    var canUseRecoveryCode: Boolean = false,
    var step: SignupStep = SignupStep.START,
    var email: String? = null,
    var loginWithRc: LoginWithRc? = null
)

class LoginWithRc(val rc: String = "", var keysetFetchNeeded: Boolean = false)
